### ISSUE TITLE
Mulled container resolution not using latest build of mulled container

### ISSUE THREAD

**Issue by pcm32** [galaxyproject/galaxy#8454]
I recently updated (a few days ago) a bioconda recipe which was merged and led to a new tag for container [seurat-scripts](https://quay.io/repository/biocontainers/seurat-scripts?tab=tags). However, today, when executing a job on Galaxy, the mulled container resolver still picks the older tag (`0.0.5--r341_0`) and not the newer tag (`0.0.5--r34_1`):

```
galaxy.tools.deps.containers INFO 2019-08-13 15:40:06,842 [p:423,w:1,m:0] [KubernetesRunner.work_thread-0] Checking with container resolver [MulledDockerContainerResolver[namespace=biocontainers]] found description [ContainerDescription[identifier=quay.io/biocontainers/seurat-scripts:0.0.5--r341_0,type=docker]]
```

Thanks!

**Comment by mvdbeek** [comment]
Do you get the new one if you remove the old image ?

**Comment by pcm32** [comment]
Mmm... I would need to check each of the nodes of the k8s I guess... I haven't run though the old one of a while, so I would expect the garbage collection in k8s to get rid of the old one. Is that what you mean, right?

**Comment by pcm32** [comment]
This was a result of a re-run (circling arrows button). Maybe that is part of the reason? will try a fresh run...

**Comment by mvdbeek** [comment]
Yes, rerun won't change anything.

**Comment by pcm32** [comment]
Yes, a fresh run picks up the correct container. I wonder if you could force this behaviour. I can imagine a situation where you want to do re-runs using the newer build of the container (provided that the underlying conda packages are the same).

**Comment by pcm32** [comment]
No, I was wrong. Fresh runs are also using the older container. The new container tag never gets picked up. I notice that the tag format changed a bit, could this be the reason?

**Comment by pcm32** [comment]
@bgruening could this be an issue at the conda side given the change in the tag format changing from `r341` to `r34` and then the build number?

**Comment by pcm32** [comment]
It seems to me that this part of the code https://github.com/galaxyproject/galaxy/blob/f6011bab5b8adbabae4986a45849bb9158ffc8bb/lib/galaxy/tool_util/deps/container_resolvers/mulled.py#L245-L249
is just picking the first build available for the specified version. I guess that there is a sorting logic somewhere upstream that would sort by version and build so that the highest build would be picked for the same version, and somehow the change in the tag (`r341` to `r34`) breaks this?

**Comment by mvdbeek** [comment]
yup, there is sorting logic in https://github.com/galaxyproject/galaxy/blob/98a8dd880e0cd27832514924e0d4845daf63b023/lib/galaxy/tool_util/deps/mulled/util.py#L76.
I assume the major issue is that r341 sorts above r34. We may be able to use the update timestamp on quay to sort build tags.
